### PR TITLE
Add function comment

### PR DESCRIPTION
**Pull request by ukos-git**
Pre-Function Comment is added as tooltip on igorpro 8

### CODE REVIEW

File: procedures/CodeBrowser.ipf [hunk 1]
```diff
@@ -878,19 +951,67 @@ Function/S nicifyProcedureList(list)
...
+	return StringFromList(lineNo, context, "\r")
+End
+
+Function/S TrimFunctionComment(context)
+	string context
+
+	string re, comment
+
+	// help for regex on https://regex101.com/
+	re="^(?is)(.*)[\r\n]+\s*Function(.*)"
```
Comment (ukos-git) [inline procedures/CodeBrowser.ipf]: @t-b need help here. See https://github.com/byte-physics/igor-code-browser/issues/11#issuecomment-445964278

Comment (ukos-git) [reply procedures/CodeBrowser.ipf]: Need to exclude lines that are preceding the comment and are not comment lines.

**Comment by ukos-git** [comment]
waiting for #23.
ed503af is kept to keep this branch functional. It should be removed after merge of #23 before rebase to master

File: procedures/CodeBrowser.ipf [hunk 2]
```diff
@@ -808,17 +843,20 @@ static Function saveLoad(procedure)
...
 	endif
 
 	WAVE/T load0 = SaveWavesWave[procedure.row][0]
 	WAVE/I load1 = SaveWavesWave[procedure.row][1]
+	WAVE/T load2 = SaveWavesWave[procedure.row][2]
 	declWave[][0] = load0[p][0]
 	declWave[][1] = load0[p][1]
 	lineWave[] = load1[p]
+	helpWave[][0] = load2[p][0]
+	helpWave[][1] = load2[p][1] //@todo why not use q?
```
Comment (t-b) [inline procedures/CodeBrowser.ipf]: Using `q` sounds like a good idea or?

File: procedures/CodeBrowser.ipf [hunk 3]
```diff
@@ -943,6 +981,46 @@ Function/S getFunctionLine(lineNo, funcInfo)
...
+
+	html = ""
+	lines = ItemsInList(context, "\r")
+	for(n = 0; n < lines; n += 1)
+		line = StringFromList(n, context, "\r")
+		if(GrepString(line, "(?i)\/\/\/.*")) // Doxygen comments
+			re = "(?i)(.*)(@param(?:\[(?:in|out)\])?)\s(\w+)\s(.*)"
+			SplitString/E=(re) line, str0, str1, str2, str3
+			if(V_flag == 4)
+				sprintf line, "%s%s <b>%s</b> %s", str0, str1, str2, str3
```
Comment (t-b) [inline procedures/CodeBrowser.ipf]: sprintf is limited to x characters depending on the igor version. You have to use `+=` unfortunately. Try opening `MIES_WaveDataFolderGetters` for example.

File: procedures/CodeBrowser.ipf [hunk 4]
```diff
@@ -943,6 +981,46 @@ Function/S getFunctionLine(lineNo, funcInfo)
...
+			endif
+			re = "(?i)(.*)@(\w+)(.*)"
+			SplitString/E=(re) line, str0, str1, str2
+			if(V_flag == 3)
+				sprintf line, "%s<i>@%s</i>%s", str0, str1, str2
```
Comment (t-b) [inline procedures/CodeBrowser.ipf]: dito.

File: procedures/CodeBrowser.ipf [hunk 5]
```diff
@@ -943,6 +981,46 @@ Function/S getFunctionLine(lineNo, funcInfo)
...
+			endif
+		endif
+		html += line + "<br>"
+	endfor
+	html = RemoveEnding(html, "<br>")
+	sprintf html, "<code>%s</code>", html
```
Comment (t-b) [inline procedures/CodeBrowser.ipf]: dito.

**Comment by t-b** [comment]
Btw. I think we stay with IP6 support for the forseeable future with the package.

**Comment by ukos-git** [comment]
Let‘s keep that in mind and have some tests before releasing. I am mostly testing on IP7 snd IP8

File: procedures/CodeBrowser.ipf (hunk 2 above)
Comment (ukos-git) [reply procedures/CodeBrowser.ipf]: I am sure there was a reason for this work-around. I adopted the style from declWave, 2 lines up

File: procedures/CodeBrowser.ipf (hunk 3 above)
Comment (ukos-git) [reply procedures/CodeBrowser.ipf]: is this relevant for only one line?

File: procedures/CodeBrowser.ipf (hunk 5 above)
Comment (ukos-git) [reply procedures/CodeBrowser.ipf]: agreed.

File: procedures/CodeBrowser.ipf (hunk 3 above)
Comment (ukos-git) [reply procedures/CodeBrowser.ipf]: And line can be of undefined length. Got it.

Comment (t-b) [reply procedures/CodeBrowser.ipf]: Yes exactly.

**Comment by ukos-git** [comment]
rebased to master, included code review

commit 4908ba8 is irrelevant for this issue but fixes that the search variable is preserved on panel-closing and not set to a nullstring `""`.

**Pull request merged**